core/src/main/java/com/yahoo/oak/OakMap.java

```diff
@@ -28,60 +28,58 @@
         implements AutoCloseable, ConcurrentNavigableMap<K, V>, ConcurrentZCMap<K, V> {
 
     private final InternalOakMap<K, V> internalOakMap;
-    /*
-     * Memory manager cares for allocation, de-allocation and reuse of the internally pre-allocated
-     * memory. There can be separate memory managing algorithms for keys and values.
-     * */
-    //private final MemoryManager valuesMemoryManager;
-    //private final MemoryManager keysMemoryManager;
+
+    // Used for iterators
     private final OakTransformer<K> keyDeserializeTransformer;
     private final OakTransformer<V> valueDeserializeTransformer;
     private final Function<Map.Entry<OakScopedReadBuffer, OakScopedReadBuffer>,
             Map.Entry<K, V>> entryDeserializeTransformer;
+
     private final OakComparator<K> comparator;
 
     // SubOakMap fields
     private final K fromKey;
     private final boolean fromInclusive;
     private final K toKey;
-    private boolean toInclusive;
+    private final boolean toInclusive;
     private final boolean isDescending;
 
-    // internal constructor, to create OakMap use OakMapBuilder
-    OakMap(K minKey, OakSerializer<K> keySerializer, OakSerializer<V> valueSerializer, OakComparator<K> oakComparator,
-        int chunkMaxItems, MemoryManager vMM, MemoryManager kMM) {
-
+    private OakMap(InternalOakMap<K, V> internalOakMap,
```

Comment:
Add a comment this is only for sub-maps creation

Reply:
Used by the other constructors. Added a comment.